### PR TITLE
SLI-1521 TreeUI should be accessed only from EDT

### PR DESCRIPTION
**Pull request by eray-felek-sonarsource**
[SLI-1521](https://sonarsource.atlassian.net/browse/SLI-1521)

[SLI-1521]: https://sonarsource.atlassian.net/browse/SLI-1521?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/ui/AbstractIssuesPanel.java [hunk 1]
```diff
@@ -188,13 +189,17 @@ private OccurenceNavigator.OccurenceInfo occurrence(Tree tree, @Nullable IssueNo
...
     // relies on the assumption that a TreeNodes will always be the last row in the table view of the tree
-    var isOldTree = tree.getSelectionPath() == null;
-    var path = isOldTree ? oldTree.getSelectionPath() : tree.getSelectionPath();
-    if (path == null) {
-      return false;
-    }
-
-    return isOldTree ? fetchNextOccurenceInfo(path, oldTree) : fetchNextOccurenceInfo(path, tree);
+    final boolean[] result = new boolean[1];
+    ApplicationManager.getApplication().invokeAndWait(() -> {
```
Comment (nquinquenel) [inline src/main/java/org/sonarlint/intellij/ui/AbstractIssuesPanel.java:193]: You can use our extension `ApplicationManager.getApplication().computeInEDT`

File: src/main/java/org/sonarlint/intellij/ui/AbstractIssuesPanel.java [hunk 2]
```diff
@@ -188,13 +189,17 @@ private OccurenceNavigator.OccurenceInfo occurrence(Tree tree, @Nullable IssueNo
...
+      var isOldTree = tree.getSelectionPath() == null;
+      var path = isOldTree ? oldTree.getSelectionPath() : tree.getSelectionPath();
+      if (path == null) {
+        result[0] = false;
+      } else {
+        result[0] = isOldTree ? fetchNextOccurenceInfo(path, oldTree) : fetchNextOccurenceInfo(path, tree);
```
Comment (nquinquenel) [inline src/main/java/org/sonarlint/intellij/ui/AbstractIssuesPanel.java:199]: I think it should be done only on `fetchNextOccurenceInfo`

**Review by nquinquenel — COMMENTED** [review]
Have you tried removing this `OccurrenceNavigator` completely?

**Comment by eray-felek-sonarsource** [comment]
> Have you tried removing this `OccurrenceNavigator` completely?

Yes, nothing changes, I think we should continue with removal

File: src/main/java/org/sonarlint/intellij/ui/AbstractIssuesPanel.java [hunk 3]
```diff
@@ -185,92 +183,11 @@ private OccurenceNavigator.OccurenceInfo occurrence(Tree tree, @Nullable IssueNo
...
-  public boolean hasPreviousOccurence() {
-    var path = (tree.getSelectionPath() == null) ? oldTree.getSelectionPath() : null;
-    if (path == null) {
-      return false;
-    }
-    var node = (DefaultMutableTreeNode) path.getLastPathComponent();
-    return (node instanceof IssueNode) && !isFirst(node);
-  }
-
   private static boolean isFirst(final TreeNode node) {
```
Comment (nquinquenel) [inline src/main/java/org/sonarlint/intellij/ui/AbstractIssuesPanel.java:186]: To be removed

File: src/main/java/org/sonarlint/intellij/ui/AbstractIssuesPanel.java [hunk 4]
```diff
@@ -185,92 +183,11 @@ private OccurenceNavigator.OccurenceInfo occurrence(Tree tree, @Nullable IssueNo
       -1);
```
Comment (nquinquenel) [inline src/main/java/org/sonarlint/intellij/ui/AbstractIssuesPanel.java:183]: To remove `occurrence` as well

**Pull request merged**